From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Oliwier Miodun <naczs@blueflow.pl>
Date: Mon, 10 Jul 2023 17:59:42 +0200
Subject: [PATCH] SculkCatalyst bloom API

== AT ==
public net.minecraft.world.level.block.entity.SculkCatalystBlockEntity$CatalystListener bloom(Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/util/RandomSource;)V

diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java b/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java
index dc7d6c443af206f50d2b4ca260d64bb86b67553c..0fbead8742da8a2ff5261b747870487a218ffc32 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java
@@ -9,4 +9,20 @@ public class CraftSculkCatalyst extends CraftBlockEntityState<SculkCatalystBlock
     public CraftSculkCatalyst(World world, SculkCatalystBlockEntity tileEntity) {
         super(world, tileEntity);
     }
+
+    // Paper start - SculkCatalyst bloom API
+    @Override
+    public void bloom(@org.jetbrains.annotations.NotNull io.papermc.paper.math.Position position, int charge) {
+        com.google.common.base.Preconditions.checkNotNull(position);
+        requirePlaced();
+
+        getTileEntity().getListener().bloom(
+            world.getHandle(),
+            getTileEntity().getBlockPos(),
+            getTileEntity().getBlockState(),
+            world.getHandle().getRandom()
+        );
+        getTileEntity().getListener().getSculkSpreader().addCursors(io.papermc.paper.util.MCUtil.toBlockPos(position), charge);
+    }
+    // Paper end
 }
